shared/configs.ts

```diff
@@ -38,31 +39,47 @@ export function isGeneralConfig(config: FlatConfigItem) {
 export function matchFile(
   filepath: string,
   configs: FlatConfigItem[],
-  ignoreOnlyConfigs: FlatConfigItem[],
+  configArray: ConfigArray,
 ): MatchedFile {
-  const globalIgnored = ignoreOnlyConfigs.flatMap(config => getMatchedGlobs(filepath, config.ignores!))
-  if (globalIgnored.length) {
-    return {
-      filepath,
-      globs: globalIgnored,
-      configs: [],
-    }
-  }
-
   const result: MatchedFile = {
     filepath,
     globs: [],
     configs: [],
   }
+
   configs.forEach((config, index) => {
     const positive = getMatchedGlobs(filepath, config.files || [])
     const negative = getMatchedGlobs(filepath, config.ignores || [])
-    if (!negative.length && positive.length)
+    if (configArray && !configArray.isFileIgnored(filepath) && positive.length > 0) {
       result.configs.push(index)
-    result.globs.push(
-      ...positive,
-      ...negative,
-    )
+      // push positive globs only when there are configs matched
+      result.globs.push(...positive)
+    }
+    // push negative globs except for unignore globs
+    result.globs.push(...negative.filter(glob => !glob.startsWith('!')))
```

Comment:
I think you might also want to push negative unignore globs here, basically all matched globs. There's a [comment here](https://github.com/eslint/config-inspector/pull/109#discussion_r1873446942) where I made that switch, and I think including them causes the UI to be more clear to the user since it indicates that the globs matched, and the user can then interpret the "matched-unignore" (vs the unclear/inconsistent greyed out ui that implies it didn't match glob-wise)